_rules/inline-link-in-paragraph-has-distinguishing-style-not-based-on-color-alone-88407d.md

```diff
@@ -0,0 +1,316 @@
+---
+id: 88407d
+name: Inline link has distinguishable style or content not based on color alone
+rule_type: atomic
+description: |
+  This rule checks that inline links are distinguishable from the surrounding text through a difference in style not based on color alone or have content that indicates they are links.
+accessibility_requirements:
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], for which all the following is true:
+
+- **link text**: the element has [visible][] [text nodes][text node] as [descendants][descendant] in the [flat tree][]; and
+- **non-link line text**: the element is [rendered on a line][] containing [visible][] [text nodes][text node] that are not [descendants][descendant] in the [flat tree][] of a [semantic link][]; and
+- **different color**: the element's [foreground color][] and the [foreground color][] of the **non-link line text** elements have a [highest possible contrast][] that is at least 3.0:1, or the element's [background color][] and the [background color][] of the **non-link line text** elements have a [highest possible contrast][] that is at least 3.0:1.
+
+## Expectation
+
+For each test target, at least one of the following is true:
+
+- **distinguishing style**: the element has a [distinguishing style][] not based on color alone from the **non-link line text** elements; or
+- **distinguishing content**: element has content (such as an image or text), located in the proximity of the element (i.e. there is no other content visually in between this content and the target element), that indicates the test target is a link.
```

Comment:
🤔 and follow-up to https://github.com/act-rules/act-rules.github.io/pull/1010#discussion_r448370779

It is possible to pass this rule without passing the other (focus/hover). Maybe, if there is a distinguishing style by default that disappear when the link is focused/hovered (I'm not really sure that would pass the rule). Would that pass the SC? I don't know. I would say this is at least bad practice and that the extra (non-colour) info should no disappear (at least not without being replaced).

We do need the special case for hover/focus because of the annoying G183.

1. If we consider that a link with distinguishing style that disappears on focus/hover is bad, then we should ditch this rule and only keep the other one (distinguishing on focus/hover). (+maybe note that it includes permanent cues)
2. If we consider that a link with distinguishing style that disappears on focus/hover is OK, then this rule should specifically mention that it only checks when the link is neither focused or hovered.

Reply:
We have [G182](https://www.w3.org/WAI/WCAG21/Techniques/general/G182) saying the SC is satisfied if this happens without the link being focused/hovered.
> For example, scattered words in text may be hypertext links that are marked as such by being printed in a different color. This technique describes a way to provide cues in addition to color so that users who may have difficulty perceiving color differences or have low vision can identify them.

And we have [G183](https://www.w3.org/WAI/WCAG21/Techniques/general/G183) saying the SC is satisfied it this only happens when the link is focused/hovered. 
> For example, scattered words in text may be hypertext links that are identified only by a difference in color with surrounding text. This technique describes a way to provide additional cues on hover and focus so that users who may have difficulty perceiving color differences or have low vision can identify them.

I don't think we should ditch either one of the rules. I also agree that it is a bad practice to make the distinguishing style disappear on focus/hover, but I'm not sure it would fail the SC...

Reply:
> I don't think we should ditch either one of the rules. I also agree that it is a bad practice to make the distinguishing style disappear on focus/hover, but I'm not sure it would fail the SC...

I agree with that.
I suggest to add "when it's neither [focused] or [hovered]" in the expectation to make is clear (and to make sure that both rules are needed because "always" implies "when focused or hovered").
Maybe with an assumption saying that we assume it's OK for WCAG if the distinguishable style disappears on hover/focus, but it's nonetheless a bad idea and should be avoided.